x-pack/legacy/plugins/monitoring/public/views/no_data/controller.js

```diff
@@ -12,32 +12,76 @@ import {
   startChecks
 } from 'plugins/monitoring/lib/elasticsearch_settings';
 import { ModelUpdater } from './model_updater';
-import { render, unmountComponentAtNode } from 'react-dom';
 import { NoData } from 'plugins/monitoring/components';
 import { timefilter } from 'ui/timefilter';
 import { I18nContext } from 'ui/i18n';
 import { CODE_PATH_LICENSE } from '../../../common/constants';
+import { MonitoringViewBaseController } from '../base_controller';
+import { i18n } from '@kbn/i18n';
 
-const REACT_NODE_ID_NO_DATA = 'noDataReact';
+export class NoDataController extends MonitoringViewBaseController {
 
-// NoDataController watches all this attributes.
-// After consulting with @Chris Roberson, decided to move this out of the class for now.
-let timeUpdateSubscription;
+  constructor($injector, $scope) {
+    const monitoringClusters = $injector.get('monitoringClusters');
+    const kbnUrl = $injector.get('kbnUrl');
+    const $http = $injector.get('$http');
+    const checkers = [
+      new ClusterSettingsChecker($http),
+      new NodeSettingsChecker($http)
+    ];
 
-export class NoDataController {
+    let watcherSet;
 
-  constructor($injector, $scope) {
-    const $executor = $injector.get('$executor');
-    this.enableTimefilter($executor);
-    this.registerCleanup($scope, $executor);
+    const getData = async () => {
+      let catchReason;
+      try {
+        const monitoringClustersData = await monitoringClusters(undefined, undefined, [CODE_PATH_LICENSE]);
+        if (monitoringClustersData && monitoringClustersData.length) {
+          kbnUrl.redirect('/home');
+          return monitoringClustersData;
+        }
+      }
+      catch (err) {
+        if (err && err.status === 503) {
+          catchReason = {
+            property: 'custom',
+            message: err.data.message,
+          };
+        }
+      }
+
+      const { updateModel } = new ModelUpdater($scope, this);
```

Comment:
Same thing here. Do we need to recreate this for every `getData()` call?

Reply:
Same here, no `this` without `super()`